src/main/java/org/prebid/server/settings/EnrichingApplicationSettings.java

```diff
@@ -24,33 +28,51 @@ public class EnrichingApplicationSettings implements ApplicationSettings {
     private final boolean enforceValidAccount;
     private final double logSamplingRate;
     private final ApplicationSettings delegate;
-    private final PriceFloorsConfigResolver priceFloorsConfigResolver;
+    private final PriceFloorsConfigValidator priceFloorsConfigValidator;
     private final JsonMerger jsonMerger;
-
     private final Account defaultAccount;
 
     public EnrichingApplicationSettings(boolean enforceValidAccount,
                                         double logSamplingRate,
-                                        Account defaultAccount,
+                                        String defaultAccountConfig,
                                         ApplicationSettings delegate,
-                                        PriceFloorsConfigResolver priceFloorsConfigResolver,
-                                        JsonMerger jsonMerger) {
+                                        PriceFloorsConfigValidator priceFloorsConfigValidator,
+                                        JsonMerger jsonMerger,
+                                        JacksonMapper mapper) {
 
         this.enforceValidAccount = enforceValidAccount;
         this.logSamplingRate = logSamplingRate;
         this.delegate = Objects.requireNonNull(delegate);
         this.jsonMerger = Objects.requireNonNull(jsonMerger);
-        this.priceFloorsConfigResolver = Objects.requireNonNull(priceFloorsConfigResolver);
-        this.defaultAccount = Objects.requireNonNull(defaultAccount);
+        this.priceFloorsConfigValidator = Objects.requireNonNull(priceFloorsConfigValidator);
+
+        this.defaultAccount = parseAccount(defaultAccountConfig, mapper);
```

Comment:
Also, please validate `defaultAccount` and fail application startup on any exception.
```
this.defaultAccount = parseAccount(defaultAccountConfig, mapper);
PriceFloorsConfigValidator.validatePriceFloorConfig(this.defaultAccount);
```

Reply:
@CTMBNara it's not how it works now, meaning that currently we'll fail only when the account doesn't have valid values and the default account has invalid values, in other words, it's not required the default account is fully valid to work (and we can break an existing config with those changes)

I have an idea how to make it better, I'll re-write the code a little bit